### DISCUSSION TITLE
How to fit a DetailedList widget in BOTH vertical and horizontal space

### DISCUSSION THREAD

**Discussion by shula** [beeware/toga#2680] [General]
details + screenshots:
https://stackoverflow.com/questions/78682977/how-to-fit-detailedview-widget-both-to-row-and-column-height-toga-python-be

in short:  
I have 3 boxes (top: fixed height, bottom: fixed height, center - fit has much as possible).
I can make the center element fit either vertcally, or horizonatally, but not both.

I have put the elements insided 3 boxes;
I assigned them "flex=1, flex=6, flex=1"

The central element either expands vertically (with direction=ROW)  or horizontally (when direction=COLUMN).
How can I make it expand both ways?

```
class MyApp(toga.App):

    def startup(self):

        data = [ ['a','b'] for _ in range(100) ]

        query = TextInput()
        table = DetailedList(  data=data  )
        button = Button("butt")
        top =       Box(style=Pack(direction=COL, flex=1, background_color='blue'), children=[query])
        center =    Box(style=Pack(direction=ROW, flex=6, background_color='red'),  children=[table])
        bottom =    Box(style=Pack(direction=ROW, flex=1, background_color='green'), children=[button])

        main_box =  Box(style=Pack(direction=COL, padding=9),  
                            children=[top, center, bottom])

        self.main_window = toga.MainWindow(title='myapp')
        self.main_window.content = main_box
        self.main_window.show()
```

**Comment by mhsmith** [comment]
If you want the DetailedList to expand, then you need to set `flex` on it directly.

Also, unless you're planning to add something else to the row, the `center` box serves no purpose, so you might as well add the DetailedList to the `main_box` directly, without wrapping it in a box.

You asked this question in a very confusing way, with half of the information on StackOverflow and half of it here. In future, please put all the information in one place.

**Reply by shula** [reply]
got it. thanks!
